File: Assignment 1/Stack.py
maxsize = 500

class Stacks:
    def __init__(self):
        self.array = []

    def Push(self,x):
        if (len(self.array) < maxsize):
            self.array.append(x)
            return 0
        else:
            print("Stack Overflow")
            return 2

    def Top(self):
        return self.array[len(self.array)-1]

    def IsEmpty(self):
        if len(self.array) == 0:
            print("**Stack is Empty**")
            return 1
        else:
            return 0
    
    def Pop(self):
        if self.IsEmpty() == 0:
            popped = self.array.pop(len(self.array)-1)
            return popped
        else:
            print("Stack Empty")
            return 1

File: Assignment 1/stack.cpp
#include <iostream>

const int maxsize = 500;
class Stacks
{
    public:
    double stack[maxsize];
    int n = -1;
    bool Push(int x, double stack[])
    {
        if (n >= (maxsize - 1))
        {
            cout << "Stack Overflow";
            return false;
        }
        else
        {
            stack[++n] = x;
        }
        return true;
    }
    bool isEmpty(double stack)
    {
        if (n == 0)
        {
            cout << "Stack is Empty"
            return true;
        }
        else return false;
    }
};